src/Microsoft.AspNet.Razor/Common/HashCodeCombiner.cs

```diff
@@ -49,7 +49,8 @@ public HashCodeCombiner Add(object o)
 
         public HashCodeCombiner Add<TValue>(TValue value, IEqualityComparer<TValue> comparer)
         {
-            return Add(comparer.GetHashCode(value));
+            var hashCode = value != null ? comparer.GetHashCode(value) : 0;
+            return Add(0);
```

Comment:
also, if the comparer can't handle `null` suggest making a similar change to `Add(object)`